stwcs/updatewcs/astrometry_utils.py

```diff
@@ -360,12 +410,252 @@ def getObservation(self, observationID):
+        hlet_extns = headerlet.get_headerlet_kw_names(obsname, kw='EXTVER')
+        # newly processed data will not have any hlet_extns, so we need to account for that
+        newhlt = max(hlet_extns) + 1 if len(hlet_extns) > 0 else 1
+        newext = max(headerlet.find_headerlet_HDUs(obsname, strict=False))
+        hlet_names = [obsname[('hdrlet', e)].header['wcsname'] for e in hlet_extns]
+
+        if wname not in hlet_names:
+            hdrname = "{}_{}".format(filename.replace('.fits', ''), wname)
+            # Create full filename for headerlet:
+            hfilename = "{}_hlet.fits".format(hdrname)
+            logger.info("Archiving pipeline-default WCS {} to {}".format(wname, filename))
+            descrip = "Pipeline-default WCS"
+            numext = len(obsname)
+            headerlet.archive_as_headerlet(obsname, hfilename,
+                                           sciext='SCI',
+                                           wcskey="PRIMARY",
+                                           author="stwcs.updatewcs",
+                                           descrip=descrip)
+            obsname[numext].header['EXTVER'] = newhlt
+
+            # Now, write out pipeline-default WCS to a unique headerlet file
+            logger.info("Writing out pipeline-default WCS {} to headerlet file: {}".format(wname, hfilename))
+            headerlet.extract_headerlet(obsname, hfilename, extnum=numext, clobber=True)
+
+        # We need to create new apriori WCS based on new IDCTAB
+        # Get guide star offsets from DB
+        # Getting observationID (rootname) from header to avoid
+        # potential issues with actual filename being changed
+        pix_offsets = self.find_gsc_offset(obsname)
+
+        # Determine rootname for IDCTAB
+        idctab = obsname[0].header['IDCTAB']
+        idcroot = os.path.basename(fileutil.osfn(idctab)).split('_')[0]
+        # Create WCSNAME for this new a priori WCS
+        wname = 'IDC_{}-{}'.format(idcroot, pix_offsets['catalog'])
+        # Compute and add new solution if it is not already an alternate WCS
+        # Save this new WCS as a headerlet extension and separate headerlet file
+        hdrname = "{}_{}".format(filename.replace('.fits', ''), wname)
+        # Create full filename for headerlet:
+        hfilename = "{}_hlet.fits".format(hdrname)
+
+        # apply offsets to image using the same tangent plane
+        # which was used to compute the offsets
+        updatehdr.updatewcs_with_shift(obsname, pix_offsets['expwcs'],
+                                       hdrname=hfilename,
+                                       wcsname=wname, reusename=False,
+                                       fitgeom='rscale', rot=0.0, scale=1.0,
+                                       xsh=pix_offsets['delta_x'],
+                                       ysh=pix_offsets['delta_y'],
+                                       verbose=False, force=True)
```

Comment:
As far as I understand it the `update_wcs_with_shifts` function computes the shifts, scale,....
Aren't these already pre-computed and just need to be updated in this specific case with the apriori update?

Reply:
The `updatewcs_with_shift` function only applies the corrections provided on input through these parameters; it does not compute any new fit on it's own.  The docstring for this function in `updatehdr` describes the process it uses to apply the corrections so that people can follow how their WCSs are being updated by the code.  The new a priori solutions for a given IDCTAB will have a set of corrections (from the database using `find_gsc_offsets`) that need to be applied and the code here is calling the function to apply those corrections.